plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/model/Library.java

```diff
@@ -200,12 +203,54 @@ public synchronized void resolveDependencies() throws CoreException {
         Collection<LibraryFile> dependencies = LibraryFactory.loadTransitiveDependencies(coordinates);
         transitiveDependencies.addAll(dependencies);
       }
-      this.libraryFiles = transitiveDependencies;
+      
+      this.libraryFiles = resolveDuplicates(transitiveDependencies);
       this.resolved = true;
     }
-  }
+  }  
   
   /**
+   * Strip out different versions of the same library, retaining only the most recent.
+   *
+   * @return a new list containing the most recent version of each dependency
+   */
+  public static List<LibraryFile> resolveDuplicates(List<LibraryFile> dependencies) {
+    List<LibraryFile> result = new ArrayList<>();
+    Map<String, MavenCoordinates> map = new HashMap<>();
+    for (LibraryFile file : dependencies) {
+      MavenCoordinates coordinates = file.getMavenCoordinates();
+      String key = coordinates.getGroupId() + ":" + coordinates.getArtifactId();
+      if (map.containsKey(key)) {
+        MavenCoordinates previousCoordinates = map.get(key);
+        if (newer(coordinates, previousCoordinates)) {
+          map.put(key, coordinates);
+          for (int i = 0; i < result.size(); i++) {
+            if (result.get(i).getMavenCoordinates().equals(previousCoordinates)) {
+              result.set(i, file); // replace
```

Comment:
The `result` list seems a bit complicated.  We seem to assume that there's a 1-1 relationship between `LibraryFile` and `MavenCoordinates` (or at least after running through this list, any `LibraryFile`s with these coordinates will be changed to provide this `file` instance, and so we'll have duplicates).  So can't we just have the `map` be `<String,LibraryFile>` instead?

Reply:
Done.